releasing/releasing/releasing.go

```diff
@@ -0,0 +1,478 @@
+	mod := module{
+		name: modName,
+		path: path,
+	}
+	mod.UpdateCurrentVersion()
+	v := mod.version.String()
+	logDebug("module %s version.toString => %s", mod.name, v)
+	return v
+}
+
+// === module version struct and functions definition ===
+
+type moduleVersion struct {
+	major int
+	minor int
+	patch int
+}
+
+func (v moduleVersion) String() string {
+	return fmt.Sprintf("v%d.%d.%d", v.major, v.minor, v.patch)
+}
+
+func (v *moduleVersion) Set(major, minor, patch int) {
+	v.major = major
+	v.minor = minor
+	v.patch = patch
+}
+
+func (v *moduleVersion) FromString(vs string) {
+	versions := strings.Split(vs, ".")
+	major, err := strconv.Atoi(versions[0])
+	if err != nil {
+		logFatal(err.Error())
+	}
+	minor, err := strconv.Atoi(versions[1])
+	if err != nil {
+		logFatal(err.Error())
+	}
+	patch, err := strconv.Atoi(versions[2])
+	if err != nil {
+		logFatal(err.Error())
+	}
+	v.Set(major, minor, patch)
+}
+
+func (v *moduleVersion) Bump(t string) {
+	if t == "major" {
+		v.major++
+		v.minor = 0
+		v.patch = 0
+	} else if t == "minor" {
+		v.minor++
+		v.patch = 0
+	} else if t == "patch" {
+		v.patch++
+	} else {
+		logFatal("Invalid version type: %s", t)
+	}
+}
+
+// === module struct and functions definition ===
+
+type module struct {
+	name    string
+	path    string
+	version moduleVersion
+}
+
+func (m *module) UpdateCurrentVersion() {
+	logDebug("Getting latest tag for %s", m.name)
+	cmd := exec.Command("git", "tag", "-l")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	cmd.Dir = m.path
+	err := cmd.Run()
+	if err != nil {
+		logFatal(err.Error())
+	}
+
+	// Search for module tag
+	regString := fmt.Sprintf("(?m)^%s/v(\\d+\\.){2}\\d+$", m.name)
+	reg := regexp.MustCompile(regString)
+	tagsString := reg.FindAllString(out.String(), -1)
+	logDebug("Tags for module %s:\n%s", m.name, tagsString)
+	var versions []moduleVersion
+	for _, tag := range tagsString {
+		tag = tag[len(m.name)+2:]
+		v := moduleVersion{}
+		v.FromString(tag)
+
+		versions = append(versions, v)
+	}
+	// Sort to find latest tag
+	sort.Slice(versions, func(i, j int) bool {
+		if versions[i].major == versions[j].major && versions[i].minor == versions[j].minor {
+			return versions[i].patch > versions[j].patch
+		} else if versions[i].major == versions[j].major {
+			return versions[i].minor > versions[j].minor
+		} else {
+			return versions[i].major > versions[j].major
+		}
+	})
+
+	m.version = versions[0]
+}
+
+func (m *module) Tag() string {
+	return m.name + "/" + m.version.String()
+}
+
+func (m module) RunTest() (string, error) {
+	if !doTest {
+		logInfo("Tests disabled.")
+		return "", nil
+	}
+	testPath := path.Join(m.path, m.name)
+	logInfo("Running tests in %s...", testPath)
+	cmd := exec.Command("go", "test", "./...")
+	cmd.Dir = testPath
+	stdoutStderr, err := cmd.CombinedOutput()
+	if err != nil {
+		return string(stdoutStderr), err
+	}
+	logInfo("Tests are successfully finished")
+	return "", nil
+}
+
+// === Git environment functions ===
```

Comment:
suggest another file, `gitrunner.go` with methods on an object like
```
type gitRunner struct {
  tmpDir string
}
```
func MakeGitRunner() *gitRunner {
  // make the struct and the associated tmp dir
}

breaking this up so we have
```
main.go  // cobra stuff
gitrunner.go  // git exec calls
modulemeta.go  // abstraction of module metadata
moduleversion.go  // abstraction of a version
```
easier to understand, and tests have a more natural home